libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1713,6 +1713,45 @@ void GCS_MAVLINK::send_accelcal_vehicle_position(uint32_t position)
     }
 }
 
+
+float GCS_MAVLINK::vfr_hud_airspeed() const
+{
+    // because most vehicles don't have airspeed sensors, we return a
+    // different sort of speed estimate in the relevant field for
+    // comparison's sake.
+    return AP::gps().ground_speed();
+}
+
+float GCS_MAVLINK::vfr_hud_climbrate() const
+{
+    return -vfr_hud_velned.z;
+}
+
+void GCS_MAVLINK::send_vfr_hud()
+{
+    AP_AHRS &ahrs = AP::ahrs();
+
+    // return values ignored; we send stale data
+    ahrs.get_position(global_position_current_loc);
+    ahrs.get_velocity_NED(vfr_hud_velned);
+
+    float alt;
+    if (vfr_hud_make_alt_relative()) {
+        ahrs.get_relative_position_D_home(alt);
+        alt = -alt;
+    } else {
+        alt = global_position_current_loc.alt / 100.0f;
```

Comment:
Well, that would certainly be very nice. If you can get buy-in from all of the vehicle and GCS maintainers, I'd *love* to make that change!

Reply:
... wasn't the real concern OSDs?

Reply:
Vehicle and GCS maintainers seems easy, there is no good reason why they would say no.

Regarding OSDs:

- alceosd seems to expect MSL: https://github.com/ArduPilot/alceosd/blob/master/firmware/alce-osd.X/flight_stats.c#L58-L60

- minimosd-extra is confusing:
  - at https://github.com/night-ghost/minimosd-extra/blob/master/MinimOsd_Extra/protocols/MAVLink.h#L313 it says it expects MSL
  - but at https://github.com/night-ghost/minimosd-extra/blob/master/MinimOsd_Extra/Panels.h#L335 it seems to use it only for Copter (where it sends altitude above home).

  @night-ghost can you please clarify?

Reply:
+1 to always doing the non relative here. Sending relative is wrong to the spec, and a very surprising vehicle difference. (And since we've been going for being spec correct recently I guess breaking it is reasonable).